### ISSUE TITLE
It doesn't work with Frame

### ISSUE THREAD

**Issue by luisivan** [gnosis/MultiSigWallet#272]
It just doesn't work with [Frame](https://frame.sh), is there any MetaMask-specific code in there? Otherwise it should just work like any web3 provider

**Comment by denisgranha** [comment]
Hello Luis, I tried to use that app but it seems to need a Hardware key and I don't have one in my hand right now so I will try tomorrow.

In general it should work with any web3 provider. When exactly doesn't work? already at injection? No error displayed on the console?  Do you know if they have the provider open source? I didn't find it in any repo.

**Comment by luisivan** [comment]
It just stays like this, without any error on the console 
![image](https://user-images.githubusercontent.com/718208/58159052-fda85580-7c7b-11e9-95de-6216ce5b7b21.png)

**Comment by denisgranha** [comment]
Hello @luisivan 

Taking a deeper look in the code, we found that the main break point is the web3 provider injected, Frame is using [web3 v1.0.0-36beta](https://github.com/floating/frame/blob/master/package.json#L117) but this introduces a few breaking changes, like the web3 batch features. I don't think this is a change without more breaking changes.

Is there a way for Frame to use last web3 stable? v0.20.7...

Best,

**Comment by luisivan** [comment]
Yeah, I don't think telling them to downgrade to such an old version makes a lot of sense... will close this for now and just keep using MetaMask

**Comment by luisivan** [comment]
Apparently Frame should work fine with 0.2, and "web3": "1.0.0-beta.36" is a devDep only used for testing. So that cannot be the issue

**Comment by denisgranha** [comment]
I don’t have experience with eth-provider but more the metamask provider engine


The first point where app breaks is by using batch requests https://github.com/ethereum/wiki/wiki/JavaScript-API#batch-requests

Then the case might be that the batch implementation/interface differs

**Comment by luisivan** [comment]
Maybe the multisig UI is relaying on these web3 exceptions that only MetaMask does https://github.com/MetaMask/faq/blob/master/DEVELOPERS.md#dizzy-all-async---think-of-metamask-as-a-light-client

**Comment by denisgranha** [comment]
![image (2)](https://user-images.githubusercontent.com/4806080/59516708-3a095300-8ec2-11e9-9d4f-2b6491cf4fdc.png)

**Comment by denisgranha** [comment]
I imagine the provider doesn't handle batches. Array of json-rpc calls

**Comment by floating** [comment]
Hey @denisgranha! `eth-provider` should support batched requests https://github.com/floating/ethereum-provider/blob/master/index.js#L107

**Comment by denisgranha** [comment]
Interesting,

You see in the log the error is triggered because _send doesn’t exist inside the sendAsync function.

Do you have any idea what could it be @floating ?

**Comment by denisgranha** [comment]
We pushed a version to https://wallet.gnosis.pm so it's easier for you to debug what might happen, as you have more expertise in the provider itself @floating 

Please let us know if we can help with something.

I will take a deeper look next week if I find something too

Best,

**Comment by luisivan** [comment]
@floating @denisgranha any progress? Thx!

**Comment by denisgranha** [comment]
So far I didn't find something, I looks that is the ethereum-provider https://github.com/floating/ethereum-provider/blob/master/index.js#L107

So far this project was tested only with metamask provider, standard web3 provider and safe provider.

But would like to have it working too with Frame, would need more insights from @floating now that we increased the logs in the app

**Comment by floating** [comment]
Hi all, sorry about the delay. I will look into this issue.

**Comment by floating** [comment]
I've tracked down the issue, it's 100% my fault. The published version of the Frame extension wasn't updated to the latest version of `ethereum-provider` (which handles batching) and I thought it had been (obviously). I've updated the extension to use the latest `ethereum-provider` and everything seems to be working now. The updated should be live in extension stores shortly. Very sorry about the confusion!
